lms/djangoapps/mobile_api/middleware.py

```diff
@@ -0,0 +1,128 @@
+"""
+Middleware for Mobile APIs
+"""
+from datetime import datetime
+from django.core.cache import cache
+from django.http import HttpResponse
+from pytz import UTC
+from mobile_api.mobile_platform import MobilePlatform
+from mobile_api.models import AppVersionConfig
+from mobile_api.utils import parsed_version
+from openedx.core.lib.mobile_utils import is_request_from_mobile_app
+from request_cache.middleware import RequestCache
+
+
+class AppVersionUpgrade(object):
+    """
+    Middleware class to keep track of mobile application version being used
+    """
+    LATEST_VERSION_HEADER = 'EDX-APP-LATEST-VERSION'
+    UPGRADE_DEADLINE_HEADER = 'EDX-APP-UPGRADE-DATE'
+    NO_LAST_SUPPORTED_DATE = 'NO_LAST_SUPPORTED_DATE'
+    NO_LATEST_VERSION = 'NO_LATEST_VERSION'
+    USER_APP_VERSION = 'USER_APP_VERSION'
+
+    def process_request(self, request):
+        """
+        raises HTTP Upgrade Require error if request is from mobile native app and
+        user app version is no longer supported
+        """
+        version_data = self.get_info_for_version(request)
+        if version_data:
+            last_supported_date = version_data.get(self.UPGRADE_DEADLINE_HEADER, None)
+            if last_supported_date and last_supported_date != self.NO_LAST_SUPPORTED_DATE:
```

Comment:
Since `version_data` must have the key, there is no need for the default `None` to the `get` call. In fact that may mask issues with `version_data.` Similarly the `if last_supported_date` check is also redundant because the value is either going to be a date or the string "NO_LAST_SUPPORTED_DATE". This also applies to the `process_response`.